react/ImageSlider.tsx

```diff
@@ -0,0 +1,233 @@
+    showPaginationDots: 'mobileOnly' | 'desktopOnly' | 'always' | 'never'
+    usePagination: boolean
+    infinite: boolean
+  }
+}
+
+const ImageSlider: StorefrontFunctionComponent<Props> = ({
+  images,
+  height,
+  sliderLayoutConfig = {
+    itemsPerPage: {
+      desktop: 1,
+      tablet: 1,
+      phone: 1,
+    },
+    infinite: true,
+    showNavigationArrows: 'always',
+    showPaginationDots: 'always',
+    usePagination: true,
+  },
+}) => {
+  const { isMobile } = useDevice()
+
+  return (
+    <SliderLayout {...sliderLayoutConfig} totalItems={images.length}>
+      {images.map(({ link, image, mobileImage, description }, idx) => (
+        <Image
+          key={idx}
+          src={isMobile && mobileImage ? mobileImage : image}
+          alt={description}
+          link={link}
+          maxHeight={height}
+          shouldFillWidth
+        />
+      ))}
+    </SliderLayout>
+  )
+}
+
+const messages = defineMessages({
+  title: { id: 'admin/editor.image-slider.title', defaultMessage: '' },
+  description: {
+    id: 'admin/editor.image-slider.description',
+    defaultMessage: '',
+  },
+  imagesImageTitle: {
+    id: 'admin/editor.image-list.images.image.title',
+    defaultMessage: '',
+  },
+  imagesMobileImageTitle: {
+    id: 'admin/editor.image-list.images.mobileImage.title',
+    defaultMessage: '',
+  },
+  imagesImageDescription: {
+    id: 'admin/editor.image-list.images.description.title',
+    defaultMessage: '',
+  },
+  imagesImageLinkUrl: {
+    id: 'admin/editor.image-list.images.link.url.title',
+    defaultMessage: '',
+  },
+  imagesImageLinkOpenNewTab: {
+    id: 'admin/editor.image-list.images.link.openNewTab.title',
+    defaultMessage: '',
+  },
+  imagesImageLinkNoFollow: {
+    id: 'admin/editor.image-list.images.link.noFollow.title',
+    defaultMessage: '',
+  },
+  imagesImageLinkTitle: {
+    id: 'admin/editor.image-list.images.link.title.title',
+    defaultMessage: '',
+  },
+  imagesTitle: {
+    id: 'admin/editor.image-list.images.title',
+    defaultMessage: '',
+  },
+  heightTitle: {
+    id: 'admin/editor.image-list.height.title',
+    defaultMessage: '',
+  },
+  sliderTitle: {
+    id: 'admin/editor.image-slider.slider.title',
+    defaultMessage: '',
+  },
+  sliderInfinite: {
+    id: 'admin/editor.image-slider.slider.infinite',
+    defaultMessage: '',
+  },
+  sliderShowNavigation: {
+    id: 'admin/editor.image-slider.slider.showNavigation',
+    defaultMessage: '',
+  },
+  sliderShowPaginationDots: {
+    id: 'admin/editor.image-slider.slider.showPaginationDots',
+    defaultMessage: '',
+  },
+  sliderUsePagination: {
+    id: 'admin/editor.image-slider.slider.usePagination',
+    defaultMessage: '',
+  },
+})
+
+ImageSlider.schema = {
+  title: messages.title.id,
+  description: messages.description.id,
+  type: 'object',
+  properties: {
+    images: {
+      items: {
+        properties: {
+          image: {
```

Comment:
Shouldn't these strings be localizable, so these props shouldn't be in the contentSchema?